internal/helm/helm.go

```diff
@@ -122,47 +124,55 @@ func makeStack(appname string, targetDir string, data []byte, stackVersion strin
 	if err := os.MkdirAll(filepath.Join(targetDir, "templates"), 0755); err != nil {
 		return err
 	}
-	var stack interface{}
+	var stackData []byte
 	switch stackVersion {
 	case V1Beta2:
 		stackSpec := templateconversion.FromComposeConfig(rendered)
-		stack = templatev1beta2.Stack{
-			TypeMeta:   typeMeta(stackVersion),
+		stack := templatev1beta2.Stack{
+			TypeMeta: templatev1beta2.TypeMeta{
+				Kind:       "Stack",
+				APIVersion: "compose.docker.com/" + stackVersion,
+			},
 			ObjectMeta: objectMeta(appname),
 			Spec:       stackSpec,
 		}
+		templatetypes.ProcessTemplate = toGoTemplate
+		stackData, err = yaml.Marshal(stack)
```

Comment:
It's very possible I've missed something but is it not possible to split the template processing/transformation from the conversion to yaml (and therefore avoid the global)? So you'd have something like (modulo `err` handing):
```go
adjusted, err := applyToAll(stack, toGoTemplate)
stackData, err = yaml.Marshal(adjusted)
```

`applyToAll` looks from the uses here like it would need to be generic as in `applyToAll(interface{}, func(string) string) interface{}` (and `reflect` based) but I think it should still be possible to retain the underlying type so you'd have something like:
```go
adjusted, err := applyToAll(stack, toGoTemplate)
stackData, err = yaml.Marshal(adjusted.(templatev1beta2.Stack))
```
here. `applyToAll` would be such that it guarantees that the type conversion would always pass.

Reply:
It could (and was) done by going through a marshal cycle to a map[interface{}]interface{} and iterating over that. But the point of this patch is to remove all those post-processing steps.